freshmaker/config.py

```diff
@@ -57,65 +57,88 @@ def all_(*rules):
+    The module may be None.
+    """
+    from conf import config
 
-    # TestConfiguration shall only be used for running tests, otherwise...
+    config_section = 'DevConfiguration'
+    config_module = None
+
+    # TestConfiguration shall only be used for running tests
     test_env = os.environ.get('FRESHMAKER_TESTING_ENV', '').lower()
     test_executables = {'py.test', 'pytest', 'pytest.py'}
     if (os.path.basename(sys.argv[0]) in test_executables or
             test_env in ('1', 'on', 'true', 'y', 'yes')):
         config_section = 'TestConfiguration'
-        from conf import config
         config_module = config
-    # ...FRESHMAKER_DEVELOPER_ENV has always the last word
-    # and overrides anything previously set before!
-    # In any of the following cases, use configuration directly from Freshmaker
-    # package -> /conf/config.py.
 
     elif os.environ.get('FRESHMAKER_DEVELOPER_ENV', '').lower() in ('1', 'on', 'true', 'y', 'yes'):
         config_section = 'DevConfiguration'
-        if 'FRESHMAKER_CONFIG_FILE' in os.environ:
-            config_file = os.environ['FRESHMAKER_CONFIG_FILE']
-            config_module = None
-        else:
-            from conf import config
+        if 'FRESHMAKER_CONFIG_FILE' not in os.environ:
             config_module = config
 
-    # try loading configuration from file
-    if not config_module:
-        try:
-            config_module = imp.load_source('freshmaker_runtime_config',
-                                            config_file)
-        except IOError:
-            raise SystemError("Configuration file {} was not found."
-                              .format(config_file))
-
-    # finally configure Freshmaker
+    # Try getting config_section from os.environ
+    elif 'FRESHMAKER_CONFIG_SECTION' in os.environ:
+        config_section = os.environ['FRESHMAKER_CONFIG_SECTION']
+
+    # Automagically detect production environment:
+    #   - existing and readable config_file presets ProdConfiguration
+    elif os.path.exists(prod_config_file) and os.access(prod_config_file, os.O_RDONLY):
+        config_section = 'ProdConfiguration'
+
+    return (config_section, config_module)
+
+
+def get_config_module_from_file(config_file):
+    """
+    Try loading configuration module from a file
+    """
+    try:
+        config_module = imp.load_source('freshmaker_runtime_config', config_file)
+    except IOError:
+        raise SystemError("Configuration file {} was not found.".format(config_file))
+
+    return config_module
+
+
+def get_freshmaker_config(app, config_section, config_module):
+    """
+    Get Freshmaker configuration
+    """
     config_section_obj = getattr(config_module, config_section)
     conf = Config(config_section_obj)
     app.config.from_object(config_section_obj)
     return conf
 
 
+def init_config(app):
+    """
+    Configure Freshmaker
+    """
+    prod_config_file = '/etc/freshmaker/config.py'
```

Comment:
What do you think if we would create a global var PROD_CONFIG_FILE = ''/etc/freshmaker/config.py'' and after it will check if  'FRESHMAKER_CONFIG_FILE'  exists as env var, config_file = FRESHMAKER_CONFIG_FILE. Otherwise config_file = PROD_CONFIG_FILE. This way we can delete def get_config_file(prod_config_file).

Reply:
I like this suggestion, thank you